src/Ecommerce.js

```diff
@@ -0,0 +1,161 @@
+    const [productsPerPage, setProductsPerPage] = useState(9);
+    const [brandNames, setBrandNames] = useState([]);
+    const [filteredProducts, setFilteredProducts] = useState();
+    const [brands, setBrands] = useState();
+    const [sort, setSort] = useState();
+    const [totalProducts, setTotalProducts] = useState();
+
+    useEffect(() => {
+        axios.get('http://localhost:8080/products')
+            .then(response => {
+                setProducts(response.data);
+            })
+    }, []);
+
+    useEffect(() => {
+        axios.get('http://localhost:8080/brands')
+            .then(response => {
+                setBrands(response.data);
+            })
+    }, [])
+
+    const handlePrevious = () => {
+        if (currPage !== 1) {
+            setCurrPage(currPage - 1);
+        }
+    }
+    const handleNext = (max) => {
+        if(currPage !== max){
+            setCurrPage(currPage + 1);
+        }
+    }
+    const paginate = (number) => {
+        setCurrPage(number);
+    }
+
+    useEffect(() => {
+        if (products) {
+            const newProducts = [...products];
+            let filteredBrands = newProducts;
+            if(brandNames.length !== 0){
+                filteredBrands = newProducts.filter(product => {
+                    return brandNames.findIndex(brand => brand === product.brand.name) !== -1;
+                    
+                })
+            };
+            if(sort && sort === "Low to high") {
+                filteredBrands.sort((a, b) => {return a.price - b.price});
+            } else if(sort === "High to low") {
+                filteredBrands.sort((a, b) => {return b.price - a.price});
+            };
+            setTotalProducts(filteredBrands.length);
+            const filterProducts = filteredBrands.slice((currPage-1)*productsPerPage, currPage*productsPerPage);
+            setFilteredProducts(filterProducts);
+        }
+    }, [products, currPage, brandNames, sort, productsPerPage]);
+
+    const goToProduct = (id) => {
+        props.history.push('/product/' + id);
+    }
+
+    const filterByBrand = (name) => {
+    
+        const index = brandNames.findIndex(brand => brand === name);
+        let newBrands;
+        if(index === -1){
+            newBrands = [...brandNames];
+            newBrands.push(name);
+            setBrandNames(newBrands);
+        }
+        else {
+            newBrands = [...brandNames]; 
+            newBrands = newBrands.filter(brand => {
+                return brand !== name;
+            });
+            setBrandNames(newBrands);
+        };
+
+        setCurrPage(1);
+    }
+
+    const sortByPrice = (option) => {
+        setSort(option);
+        setCurrPage(1);
+    }
+
+    const changeLayout = (val) => {
+        setProductsPerPage(3*val);
+        setCurrPage(1);
+    }
+
+    return (
+        <div>
+            <main className="wrapper">
+                <div>
+                <div className="card" style={{ width: "18rem" }}>
```

Comment:
Use classname instead of style prop, use it only if you must (like if your style is depending on a variable value).

Reply:
Also in all of the rest of your pr